krab/utils.go

```diff
@@ -0,0 +1,33 @@
+package krab
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	// ErrNoSuchKey is returned if a key of the given name is not found in the store
+	ErrNoSuchKey = fmt.Errorf("no key by the given name was found")
+	// ErrKeyExists is returned when writing a key would overwrite an existing key
+	ErrKeyExists = fmt.Errorf("key by that name already exists, refusing to overwrite")
+	// ErrKeyFmt is returned when the key's format is invalid
+	ErrKeyFmt = fmt.Errorf("key has invalid format")
+)
+
+func validateName(name string) error {
```

Comment:
whats the reasoning behind these restrictions?

Reply:
:man_shrugging:  Couldn't tell you the exact reasoning, other than that it seems to be a "semi-strict" requirement of the `go-ipfs-keystore` package https://github.com/ipfs/go-ipfs-keystore/blob/508c1e10090009fde253d4d9eb9e2c581e3781ad/fskeystore.go#L54.

Ideally this should be able to be used in place of existing keystore packages, so I figured it would be best to follow what ipfs is doing.

Reply:
ah thats fair 👍